http.go

```diff
@@ -18,15 +20,28 @@ func NewHTTPClient(ctx context.Context) *http.Client {
 	return defaultClient
 }
 
-func HTTPRequestExecutor(clientFactory transport.HTTPClientFactory) transport.HTTPRequestExecutor {
+func HTTPRequestExecutor(clientFactory transport.HTTPClientFactory, cfg *config.Backend) transport.HTTPRequestExecutor {
```

Comment:
I'd create a new function `HTTPRequestExecutorFromConfig` so we don't break the interfaces and signatures exposed by the package

Reply:
The code for backend path aggregration needs the config though, otherwise it can't access the configured pattern and only the requested url.
So to prevent users of the old method from running into the cardinality issue, i think we could return a empty string as path if no config is available.